### DISCUSSION TITLE
Is it possible to use `vike` together with TS `moduleResolution: Bundler`?

### DISCUSSION THREAD

**Discussion by ChristophP** [vikejs/vike#1297] [Discussion] [answered]
Hi,

lately I have been cleaning up some TS configs and tried to set [modern resolution modes](https://www.typescriptlang.org/tsconfig#moduleResolution).

I tried `Node16` which is quite strict and it choked on missing file extensions in the code. Since everything get built by Vite I thought the new `Bundler` option would fit well since it's similar to `Node16` but doesn't require file extensions in imports.

However when using `Bundler` this error occurs.

```
./node_modules/vike/dist/esm/node/runtime/html/stream.d.ts

 error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

1 /// <reference types="node" resolution-mode="require"/>
                        ~~~~
```

So I was wondering, is it possible to use `moduleResolution: Bundler` with Vike?

**Comment by brillout** [comment]
I believe so, let me quickly check.

**Reply by brillout** [reply]
Changing [`examples/vue-full-v1/tsconfig.json#moduleResolution`](https://github.com/vikejs/vike/blob/343492dae2a343e9940eaf2e5547de672ce62499/examples/vue-full-v1/tsconfig.json#L5) to `"Bundler"` works fine. So I'm inclined to think your issue is a user land one, but minimal reproduction welcome.

**Reply by ChristophP** [reply]
Ah interesting. I'll check to see if there's anything in our config that might mess with it.
Thank you for the quick reply.

**Comment by ChristophP** [comment]
I played with the TS options a bit and compared my `tsconfig.json` to the one in the example you posted.
I had `skipLibCheck: false` set when the error above occurred. Setting it to `true` makes the error go away.
I guess I could disable it. Did you set `skipLibCheck: true` in the example for the same reason?

**Reply by brillout** [reply, marked as answer]
I always set `skipLibCheck` to `true` 🙂 (maybe I shouldn't, but so far I haven't had any issues with this strategy).

**Reply by ChristophP** [reply]
Cool, I will do so as well. Thanks a lot :-)

**Reply by brillout** [reply]
And since I haven't heard anyone else complain about this, I'm guessing that always setting `skipLibCheck` to `true` is quite a common practice.

**Reply by ChristophP** [reply]
Yeah, true. I've seen it in a lot of places, I was just hesitant since I didn't fully understood consequences. Feels a bit weird, but hey :-)
